src/items/services/item.service.ts

```diff
@@ -43,4 +43,12 @@ export class ItemService {
 
     return this.itemRepository.createItem(itemRequestDto, accountId);
   }
+
+  async listAnItem(itemId: string, price: number): Promise<Item> {
+    const item = await this.itemRepository.listAnItem(itemId, price);
```

Comment:
I see two different things going on here, one is the listing functionality, and the other one the find an item functionality. It is kind of mixing things up, since right below you are returning a not found exception. What I would do first is see if the item exist, and just once I made sure it does, trying to update it.